### ISSUE TITLE
endOf* functions behavior

### ISSUE THREAD

**Issue by NaridaL** [date-fns/date-fns#1612]
Currently, the endOf* functions return 1ms before the start of the next interval, which I find unintuitive:

It seems to me `{start: startOfMinute(now), end: endOfMinute(now)}` should return an interval with a length of one minute. (not 1 minute - 1ms)

The current definition conflicts conceptually with `areIntervalsOverlapping`, which defines two intervals which "touch" as "not overlapping".

Does anyone depend on the current behavior or find it more intuitive?

**Comment by jedwards1211** [comment]
I agree with this, I don't get why various datetime libs do endOf this way.  If anyone uses `window.performance.now()`, which has submillisecond precision, they would have bugs comparing with `endOf*`.  If `endOf*` isn't going to be changed then it should probably be deprecated and `startOfNext*` functions added.  In any case it would be good to explain to users in the `endOf*` docs why it's safer to do `< startOf*() + add*(1)` instead of `<= endOf*()`.
